core/arch/arm/plat-imx/drivers/imx_huk.c

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2020 Pengutronix, Rouven Czerwinski <entwicklung@pengutronix.de>
+ */
+#include <drivers/imx_caam_mkvb.h>
+#include <kernel/tee_common_otp.h>
+#include <string.h>
+
+static uint8_t stored_key[MKVB_SIZE];
+static bool mkvb_retrieved;
+
+TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
```

Comment:
This function can be directly part of the caam_mkvb.c file as caam_mkvb.c is built only if CFG_NXP_CAAM_MKVB_DRV enabled it.

Reply:
This will work until plat-ls wants to enable the MKVB/BLOB driver, they also implement a `tee_otp_get_hw_unique_key()` function using a TF-A platform service.

Reply:
Valid point but if a platform including CAAM doesn't need/wait to use this specific CAAM operation, the CFG_CFG_NXP_CAAM_MKVB_DRV doesn't have to be enabled in the configuration file.